client/my-sites/sites/sites.jsx

```diff
@@ -93,4 +108,13 @@ export default React.createClass( {
 			</Main>
 		);
 	}
-} );
+}
+
+export default connect(
+	( state ) => ( {
+		selectedSite: getSelectedSite( state ),
+		isPrivateSite: ( site ) => isPrivateSite( state, site.ID ),
```

Comment:
Thinking about it, I'm not sure how `mergeProps` would help. Any idea @nb ?

I guess we could always define a new intermediate function and use currying:
```
const isPrivate = state => siteId => isPrivateSite( state, siteId );

export default connect(
	( state ) => ( {
		isPrivate: isPrivate( state )
	} )
)( Sites );
```

I haven't encountered a single case where we do this in calypso though.